### PR TITLE
Reset GPIO 18 on Homegear start to support HM-MOD-RPI-PCB controller

### PR DESCRIPTION
**Pull request by mstormi**
Related: #1097 

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/packages.bash [hunk 1]
```diff
@@ -223,7 +223,7 @@ homegear_setup() {
   if ! cond_redirect install -m 644 "${BASEDIR:-/opt/openhabian}"/includes/homegear-management.service /etc/systemd/system/homegear-management.service; then echo "FAILED (copy service)"; return 1; fi
   if ! cond_redirect rm -f /lib/systemd/system/homegear*; then echo "FAILED (clean default service)"; return 1; fi
   if running_in_docker; then sed -i '/RuntimeDirectory/d' /etc/systemd/system/homegear*; fi
-  if ! zram_dependency install homegear homegear-management; then return 1; fi
+  if ! zram_dependency install homegear homegear-nodes-core homegear-management wiringpi; then return 1; fi
```
Comment (ecdye) [inline functions/packages.bash:226]: I know why you added the `wiringpi` package but why did you add the `homegear-nodes-core` package?

**Review by ecdye — CHANGES_REQUESTED** [review]
Please fix tests as they currently do not all pass.

File: functions/packages.bash (hunk 1 above)
Comment (mstormi) [reply functions/packages.bash:226]: because that's part of the homegear install instructions

**Comment by mstormi** [comment]
> 
> 
> Please fix tests as they currently do not all pass.

I have no idea why they fail. Can't imagine it's this change though.
Frankly I don't know how to debug this for the time being. Do you have any idea ?

**Comment by ecdye** [comment]
Well the test that failed IIRC was part of the homegear package which was changed by this PR. Maybe because the emulation does not know about GPIO pins and therefore fails. Fair warning that is just a wild guess.

**Comment by mstormi** [comment]
I've really forgotten how to debug that, could you give me a hint please ?


For a test to approach the cause, I've removed everything but the mod to systemd config which is the core of the PR, and BATS still fails so this seems to be the cause. That now is something like `ExecStartPre=/usr/bin/gpio export 18 out` so it accesses HW which very likely fails (can't really see the command's output) on our virtual HW.
Any idea how to handle that other than to disable the test?

File: functions/packages.bash (hunk 1 above)
Comment (mstormi) [reply functions/packages.bash]: but you're right, I have not encountered a true need so removed it again

**Comment by ecdye** [comment]
It was the `if [ "$status" -ne 0 ]; then echo "$output" >&3; fi` command we just placed extras while debugging to find errors or we would just echo the output anyway to see what was happening.

**Comment by mstormi** [comment]
Thanks. Yes it was the access to the GPIO HW that didn't work (for obvious reasons).
See how I solved that now and merge if you are ok with that.

**Review by ecdye — APPROVED** [review]
Hmmm, I feel like it might be a little unnecessary to use that solution but if you think it is best go ahead and merge. An alternative could be to use the `running_in_docker` command from the helpers file.

**Comment by mstormi** [comment]
> An alternative could be to use the `running_in_docker` command from the helpers file.

Trouble is, it's initiated from the systemd config so we would need to dynamically generate that.
And that would still not be enough, compatibility IMHO should always be validated on startup not just at install time.
Think what if someone changes HW or packages after install or uses non-RPi HW (possibly incompatible with wiringpi etc).
